lang/cplusplus/cplusplus.py

```diff
@@ -0,0 +1,1139 @@
+from talon import Context, Module, actions, settings
+import re
+
+mod = Module()
+
+ctx = Context()
+ctx.matches = r"""
+tag: user.cplusplus
+"""
+
+ctx.lists["self.cpp_pointers"] = {
+    "pointer": "*",
+    "reference": "&",
+    "R value reference": "&&",
+    "universal reference": "&&",
+    "forwarding reference": "&&",
+    # C++/CLI extensions:
+    "managed pointer": "^",
+    "tracking reference": "%",
+}
+
+# Type modifiers and qualifiers
+ctx.lists["self.cpp_type_sign_modifiers"] = {
+    "signed": "signed",
+    "unsigned": "unsigned",
+}
+ctx.lists["self.cpp_type_qualifiers"] = {
+    "const": "const",
+    "volatile": "volatile",
+}
+ctx.lists["self.cpp_type_bit_width"] = {
+    "eight": "8",
+    "sixteen": "16",
+    "thirty two": "32",
+    "sixty four": "64",
+}
+
+ctx.lists["self.cpp_simple_types"] = {
+    "auto": "auto",
+    "character": "char",
+    "char": "char",
+    "wide char": "wchar_t",
+    "char eight": "char8_t",
+    "char sixteen": "char16_t",
+    "char thirty two": "char32_t",
+    "short": "short",
+    "long": "long",
+    "long long": "long long",
+    "int": "int",
+    "integer": "int",
+    "void": "void",
+    "double": "double",
+    "long double": "long double",
+    "struct": "struct",
+    "struck": "struct",
+    "num": "enum",
+    "union": "union",
+    "float": "float",
+    "size T": "size_t",
+    "size type": "size_t",
+    "bool": "bool",
+    "boolean": "bool",
+    "typename": "typename",
+}
+
+# Words that trigger the standard namespace logic
+ctx.lists["self.cpp_standard"] = {
```

Comment:
I'm not sure whether using a list here is the right thing. You could instead use `(standard|stood)` everywhere this list is used. I don't feel strongly about this, but it is a bit unusual.

Reply:
It's meant as a customization point since everyone pronounces `std` differently (https://twitter.com/jfbastien/status/1183894210971193345)

Reply:
As someone that uses C all the time but will never use C++, I'd much rather them be separate or I guess preferably C++ be an extension in different files. It makes it easier to quickly find/extend C stuff, and as I am plagued by misrecognitions in general, the less potential mismatching the better.